sap/cli/object.py

```
@@ -42,6 +45,35 @@ def object_name_from_source_file(filesystem_path):
+
+    editor = (os.environ.get('SAPCLI_EDITOR')
+              or os.environ.get('EDITOR')
+              or os.environ.get('VISUAL')
+              or 'vi')
+
+    handle, path = tempfile.mkstemp(suffix=suffix)
+    try:
+        with os.fdopen(handle, 'w', encoding='utf8') as tmp:
+            tmp.write(content)
+
+        result = subprocess.run(shlex.split(editor) + [path], check=False)
```

Ruff (0.15.21)

[error] `subprocess` call: check for execution of untrusted input

(S603)